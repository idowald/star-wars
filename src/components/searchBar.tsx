import React from "react";
import { Field, Form } from "react-final-form";
import { createStyles, makeStyles, Theme } from "@material-ui/core/styles";

interface Props {
  search: string;
  searchAction: (submit: { search: string }) => void;
}

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    form: {
      display: "flex",
      justifyContent: "space-evenly",
      width: "20%",
      alignItems: "baseline"
    }
  })
);
export const SearchBar = ({ search, searchAction }: Props) => {
  const classes = useStyles();

  return (
    <Form
      onSubmit={searchAction}
      initialValues={{ search }}
      render={({ handleSubmit, form, submitting, pristine, values }) => (
        <form onSubmit={handleSubmit} className={classes.form}>
          <label>Search</label>
          <Field
            name="search"
            component="input"
            type="text"
            placeholder="search..."
          />
          <button
            type="submit"
            disabled={submitting || pristine || !values.search}
          >
            Search
          </button>
        </form>
      )}
    />
  );
};
