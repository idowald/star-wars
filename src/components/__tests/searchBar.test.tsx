import React from "react";
import renderer from "react-test-renderer";
import {SearchBar} from "../searchBar";

test("test searchbar for correct search term", () => {
  const searchTerm = "search me";
  const rendered = renderer.create(
      <SearchBar search={searchTerm} searchAction={()=>{}} />
  );
  const renderJson = rendered.toJSON();
  // @ts-ignore
  expect(renderJson.children[1].props.value).toEqual(searchTerm);

});
test("test searchbar display snapshot", () => {

  const rendered = renderer.create(
      <SearchBar search={"search me"} searchAction={()=>{}} />
  );
  expect(rendered).toMatchSnapshot();
});
