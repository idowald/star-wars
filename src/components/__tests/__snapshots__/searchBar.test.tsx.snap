// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test searchbar display snapshot 1`] = `
<form
  className="makeStyles-form-1"
  onSubmit={[Function]}
>
  <label>
    Search
  </label>
  <input
    name="search"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    placeholder="search..."
    type="text"
    value="search me"
  />
  <button
    disabled={true}
    type="submit"
  >
    Search
  </button>
</form>
`;
